src/state/slices/portfolioSlice/selectors.ts

```diff
@@ -372,26 +373,31 @@ export const selectPortfolioFiatBalanceByFilter = createCachedSelector(
   },
 )((_s: ReduxState, filter) => `${filter?.accountId}-${filter?.assetId}` ?? 'accountId-assetId')
 
+/**
+ * TODO(0xdef1cafe): delete this when implementing idle - use selectPortfolioCryptoHumanBalanceByFilter
+ * below which respects accountIds
+ */
 export const selectPortfolioCryptoBalanceByAssetId = createCachedSelector(
   selectPortfolioAssetBalances,
   selectAssetIdParamFromFilter,
-  (byId, assetId): string => byId[assetId] ?? 0,
+  (byId, assetId): string | undefined => assetId && byId[assetId],
 )((_s: ReduxState, filter) => filter?.assetId ?? 'assetId')
 
 export const selectPortfolioCryptoHumanBalanceByFilter = createCachedSelector(
   selectAssets,
   selectPortfolioAccountBalances,
   selectPortfolioAssetBalances,
-  selectAccountIdParamFromFilterOptional,
+  selectAccountIdParamFromFilter,
   selectAssetIdParamFromFilter,
-  (assets, accountBalances, assetBalances, accountId, assetId): string => {
+  (assets, accountBalances, assetBalances, accountId, assetId): string | undefined => {
     if (accountId && assetId) {
       return fromBaseUnit(
         bnOrZero(accountBalances?.[accountId]?.[assetId]),
         assets?.[assetId]?.precision ?? 0,
       )
     }
 
+    if (!assetId) return
     return fromBaseUnit(bnOrZero(assetBalances[assetId]), assets?.[assetId]?.precision ?? 0)
```

Comment:
early return pattern as main suggestion. extra var for reduced line breaks if you like
```suggestion
    if (!assetId) return
    const precision = assets?.[assetId]?.precision ?? 0
    if (accountId) return fromBaseUnit(bnOrZero(accountBalances?.[accountId]?.[assetId]), precision)
    return fromBaseUnit(bnOrZero(assetBalances[assetId]), precision) 
```